#### About

> Hypothesis testing for means and proportion

Hypothesis testing is a statistical method used to make decisions based on data, where we compare a null hypothesis and an alternative hypothesis. The null hypothesis represents the default assumption, while the alternative hypothesis represents the claim that we want to test.

When it comes to hypothesis testing for means and proportions, we typically use a z-test or a t-test for means and a chi-squared test for proportions. The steps for hypothesis testing include:

- State the null and alt. hypothesis.
- Determine the appropriate test statistic and level of significance
- calc. test statistics.
- Determine the p-value.
- Make decision and interpret the results



Suppose we have a sample of 20 students and we want to test whether their average test score is significantly different from 75. We can use a t-test to test this hypothesis, with a significance level of 0.05.



In [3]:
import numpy as np
from scipy.stats import ttest_1samp

# Sample data
data = np.array([72, 75, 80, 68, 70, 74, 73, 72, 76, 78, 75, 71, 79, 72, 73, 76, 70, 75, 77, 74])

# Null hypothesis: The population mean is 75
# Alternative hypothesis: The population mean is not 75
null_mean = 75
alpha = 0.05

# Calculate the test statistic and p-value
t_stat, p_value = ttest_1samp(data, null_mean)


print(t_stat,p_value)
# Make a decision and interpret the results
if p_value < alpha:
    print("Reject null hypothesis. The population mean is not 75.")
else:
    print("Fail to reject null hypothesis. The population mean is 75.")

-1.4217160742372863 0.17131650734701984
Fail to reject null hypothesis. The population mean is 75.


Here, The calculated t-statistic is -1.42 and the p-value is 0.17 which is greater than significance level of 0.05. 
Hence, We fail to reject the null hypothesis and conclude that there is not enough evidence to suggest that population mean is different from 75.